iidm/iidm-api/src/main/java/com/powsybl/iidm/network/VoltageLevel.java

```diff
@@ -358,20 +358,28 @@
         }
 
         /**
-         * Get the number of node.
+         * Get the count of used nodes (nodes attached to an equipment, a switch or an internal connection).
+         *
+         * @deprecated Depending on your use case, use {@link #getMaximumNodeIndex()} or {@link #getNodeCapacity()}.
          */
-        int getNodeCount();
+        @Deprecated
+        default int getNodeCount() {
+            throw new UnsupportedOperationException();
+        }
+
+        default int getMaximumNodeIndex() {
+            throw new UnsupportedOperationException();
+        }
+
+        default int getNodeCapacity() {
```

Comment:
Please add Javadoc on this method to well understand what is expected

Reply:
At the end, I deleted this method since it always returns `getMaximumNodeIndex() + 1`. It was useful before having `cleanVertices`.